stores/utxo/aerospike/get.go

```diff
@@ -1333,35 +1366,29 @@ func (s *Store) GetTxFromExternalStore(ctx context.Context, previousTxHash chain
 func (s *Store) getExternalTransaction(ctx context.Context, previousTxHash chainhash.Hash) (*bt.Tx, error) {
 	fileType := fileformat.FileTypeTx
 
-	// Get the raw transaction from the externalStore...
-	txBytes, err := s.externalStore.Get(
-		ctx,
-		previousTxHash[:],
-		fileType,
-	)
+	// Stream parse from external store to avoid double memory allocation (raw bytes + parsed tx)
+	// This saves ~50% memory by eliminating the intermediate txBytes buffer
+	reader, err := s.externalStore.GetIoReader(ctx, previousTxHash[:], fileType)
 	if err != nil {
 		// Try to get the data from an output file instead
 		fileType = fileformat.FileTypeOutputs
 
-		txBytes, err = s.externalStore.Get(
-			ctx,
-			previousTxHash[:],
-			fileType,
-		)
+		reader, err = s.externalStore.GetIoReader(ctx, previousTxHash[:], fileType)
 		if err != nil {
 			return nil, errors.NewStorageError("[GetTxFromExternalStore][%s] could not get tx from external store", previousTxHash.String(), err)
 		}
 	}
+	defer reader.Close()
 
 	tx := &bt.Tx{}
 
 	if fileType == fileformat.FileTypeTx {
-		tx, err = bt.NewTxFromBytes(txBytes)
-		if err != nil {
-			return nil, errors.NewTxInvalidError("[GetTxFromExternalStore][%s] could not read tx from bytes", previousTxHash.String(), err)
+		// Stream parse directly from reader - no intermediate buffer
+		if _, err = tx.ReadFrom(reader); err != nil {
+			return nil, errors.NewTxInvalidError("[GetTxFromExternalStore][%s] could not read tx from stream", previousTxHash.String(), err)
 		}
 	} else {
-		bufferedReader := bufio.NewReader(bytes.NewReader(txBytes))
+		bufferedReader := bufio.NewReader(reader)
```

Comment:
This buffer should also be used for reading the transactions. Saves a lot on syscalls.